web/src/main/java/org/cbioportal/web/util/DataBinner.java

```diff
@@ -308,11 +309,13 @@ public <T extends DataBinFilter> Collection<DataBin> calcNumericalDataBins(
             } else if (DataBinFilter.BinMethod.MEDIAN == binMethod) {
                 // NOOP - handled later
             } else if (DataBinFilter.BinMethod.QUARTILE == binMethod) {
-                List<BigDecimal> bins = Arrays.asList(
+                // Edge case: some of Q1, Q2, and Q3 are the same value.
+                // Solution: reduce bins to represent unique values only.
+                List<BigDecimal> bins = Stream.of(
```

Comment:
@Luke-Sikina Well, the existing implementation only worked when Q1, Q2 and Q3 were all different values. I practice this is not always the case. In this data series [10, 250, 250, 250, 250, 250, 250, 250] Q1, Q2 and Q3 are all 250. This would in effect mean that there is a single bin, not three. The included tests describe these situations quite ok I think.

Btw the break of logic was also caused by incorrect login for evaluating the median (see comment by @dippindots below).